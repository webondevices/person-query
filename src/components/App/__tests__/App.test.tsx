// import configureStore from 'redux-mock-store';
// import {render, shallow} from 'enzyme';
// import * as React from 'react';
// import {RootState} from '../../../reducers';
// import App from '../App';
// import * as actions from '../../../actions';

// it('renders correctly', () => {
//   const store = configureStore<Partial<RootState>>()({
//     todos: {
//       personList: ['item'],
//     },
//   });

//   const props = {
//     store,
//     color: 'red',
//   };

//   const component = <App {...props} />;
//   const wrapper = render(component);
//   expect(wrapper).toMatchSnapshot();
// });

// it('triggers addTodo action when Add button pressed', () => {
//   const store = configureStore<Partial<RootState>>()({
//     todos: {
//       personList: ['item'],
//     },
//   });

//   const props = {
//     store,
//     color: 'red',
//   };

//   const component = <App {...props} />;
//   const wrapper = shallow(component)
//     .dive()
//     .dive();

//   const addButton = wrapper.find('#addButton');
//   const deleteButton = wrapper.find('#deleteButton');

//   addButton.simulate('click');
//   deleteButton.simulate('click');

//   expect(store.getActions()).toEqual([
//     actions.addTodo('todo'),
//     actions.deleteLastTodo(),
//   ]);
// });
