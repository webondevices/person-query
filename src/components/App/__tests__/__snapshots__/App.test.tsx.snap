// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-bdVaJa kfzMnH",
    },
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "data": "Welcome to React with Typescript",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "h1",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "sc-bwzfXH bukUYK",
            "color": "red",
          },
          "children": Array [
            Object {
              "data": "The color of this page is:red",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "p",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "sc-bwzfXH bukUYK",
              "color": "red",
            },
            "children": Array [
              Object {
                "attribs": Object {},
                "children": Array [
                  Object {
                    "data": "item",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "id": "addButton",
                    "type": "button",
                  },
                  "children": Array [
                    Object {
                      "data": "Add",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "button",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "id": "deleteButton",
                      "type": "button",
                    },
                    "children": Array [
                      Object {
                        "data": "Remove",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "button",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "id": undefined,
                      "type": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "id": undefined,
                      "type": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "id": undefined,
                    "type": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "id": undefined,
                    "type": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              Object {
                "attribs": Object {
                  "id": "addButton",
                  "type": "button",
                },
                "children": Array [
                  Object {
                    "data": "Add",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "id": "deleteButton",
                    "type": "button",
                  },
                  "children": Array [
                    Object {
                      "data": "Remove",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "button",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "id": undefined,
                    "type": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "id": undefined,
                    "type": undefined,
                  },
                },
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {},
                  "children": Array [
                    Object {
                      "data": "item",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "span",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {},
                  "x-attribsPrefix": Object {},
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "id": undefined,
                  "type": undefined,
                },
                "x-attribsPrefix": Object {
                  "id": undefined,
                  "type": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "id": "deleteButton",
                  "type": "button",
                },
                "children": Array [
                  Object {
                    "data": "Remove",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "id": "addButton",
                    "type": "button",
                  },
                  "children": Array [
                    Object {
                      "data": "Add",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "button",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {},
                    "children": Array [
                      Object {
                        "data": "item",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "span",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {},
                    "x-attribsPrefix": Object {},
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "id": undefined,
                    "type": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "id": undefined,
                    "type": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "id": undefined,
                  "type": undefined,
                },
                "x-attribsPrefix": Object {
                  "id": undefined,
                  "type": undefined,
                },
              },
            ],
            "name": "p",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "color": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "color": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "color": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "color": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      Object {
        "attribs": Object {
          "class": "sc-bwzfXH bukUYK",
          "color": "red",
        },
        "children": Array [
          Object {
            "data": "The color of this page is:red",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "p",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "sc-bwzfXH bukUYK",
            "color": "red",
          },
          "children": Array [
            Object {
              "attribs": Object {},
              "children": Array [
                Object {
                  "data": "item",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "id": "addButton",
                  "type": "button",
                },
                "children": Array [
                  Object {
                    "data": "Add",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "id": "deleteButton",
                    "type": "button",
                  },
                  "children": Array [
                    Object {
                      "data": "Remove",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "button",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "id": undefined,
                    "type": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "id": undefined,
                    "type": undefined,
                  },
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "id": undefined,
                  "type": undefined,
                },
                "x-attribsPrefix": Object {
                  "id": undefined,
                  "type": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Object {
              "attribs": Object {
                "id": "addButton",
                "type": "button",
              },
              "children": Array [
                Object {
                  "data": "Add",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "id": "deleteButton",
                  "type": "button",
                },
                "children": Array [
                  Object {
                    "data": "Remove",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "id": undefined,
                  "type": undefined,
                },
                "x-attribsPrefix": Object {
                  "id": undefined,
                  "type": undefined,
                },
              },
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {},
                "children": Array [
                  Object {
                    "data": "item",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "id": undefined,
                "type": undefined,
              },
              "x-attribsPrefix": Object {
                "id": undefined,
                "type": undefined,
              },
            },
            Object {
              "attribs": Object {
                "id": "deleteButton",
                "type": "button",
              },
              "children": Array [
                Object {
                  "data": "Remove",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "id": "addButton",
                  "type": "button",
                },
                "children": Array [
                  Object {
                    "data": "Add",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {},
                  "children": Array [
                    Object {
                      "data": "item",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "span",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {},
                  "x-attribsPrefix": Object {},
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "id": undefined,
                  "type": undefined,
                },
                "x-attribsPrefix": Object {
                  "id": undefined,
                  "type": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "id": undefined,
                "type": undefined,
              },
              "x-attribsPrefix": Object {
                "id": undefined,
                "type": undefined,
              },
            },
          ],
          "name": "p",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "color": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "color": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "data": "Welcome to React with Typescript",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "h1",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "color": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "color": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "sc-bwzfXH bukUYK",
          "color": "red",
        },
        "children": Array [
          Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "data": "item",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "id": "addButton",
                "type": "button",
              },
              "children": Array [
                Object {
                  "data": "Add",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "id": "deleteButton",
                  "type": "button",
                },
                "children": Array [
                  Object {
                    "data": "Remove",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "id": undefined,
                  "type": undefined,
                },
                "x-attribsPrefix": Object {
                  "id": undefined,
                  "type": undefined,
                },
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "id": undefined,
                "type": undefined,
              },
              "x-attribsPrefix": Object {
                "id": undefined,
                "type": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          Object {
            "attribs": Object {
              "id": "addButton",
              "type": "button",
            },
            "children": Array [
              Object {
                "data": "Add",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "button",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "id": "deleteButton",
                "type": "button",
              },
              "children": Array [
                Object {
                  "data": "Remove",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "id": undefined,
                "type": undefined,
              },
              "x-attribsPrefix": Object {
                "id": undefined,
                "type": undefined,
              },
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {},
              "children": Array [
                Object {
                  "data": "item",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "id": undefined,
              "type": undefined,
            },
            "x-attribsPrefix": Object {
              "id": undefined,
              "type": undefined,
            },
          },
          Object {
            "attribs": Object {
              "id": "deleteButton",
              "type": "button",
            },
            "children": Array [
              Object {
                "data": "Remove",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "button",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "id": "addButton",
                "type": "button",
              },
              "children": Array [
                Object {
                  "data": "Add",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {},
                "children": Array [
                  Object {
                    "data": "item",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "id": undefined,
                "type": undefined,
              },
              "x-attribsPrefix": Object {
                "id": undefined,
                "type": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "id": undefined,
              "type": undefined,
            },
            "x-attribsPrefix": Object {
              "id": undefined,
              "type": undefined,
            },
          },
        ],
        "name": "p",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "sc-bwzfXH bukUYK",
            "color": "red",
          },
          "children": Array [
            Object {
              "data": "The color of this page is:red",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "p",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "data": "Welcome to React with Typescript",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "h1",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "color": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "color": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "color": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "color": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
